src/main/java/org/passay/NumberRangeRule.java

```diff
@@ -0,0 +1,135 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.passay;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Rule for determining if a password contains any number within a defined range, lower inclusive, upper exclusive.
+ *
+ * @author  Middleware Services
+ */
+public class NumberRangeRule implements Rule
```

Comment:
Actually, I need to think about this some more. The use case was preventing a calendar year from being used on the end of a password. So if you wanted to prevent numbers 1917-2117, I'm not sure you could model that as a sequence.